pkg/datastore/mysql/query.go

```diff
@@ -172,3 +226,33 @@ func refineFiltersValue(filters []datastore.ListFilter) []interface{} {
 	}
 	return filtersVals
 }
+
+// makePaginationCursorValues builds array of element values used on pagination condition check.
+func makePaginationCursorValues(opts datastore.ListOptions) ([]interface{}, error) {
+	// Skip pagination on cursor is empty.
+	if len(opts.Cursor) == 0 {
+		return nil, nil
+	}
+
+	// Decode last object of previous page stored as opts.Cursor to string.
+	data, err := base64.StdEncoding.DecodeString(opts.Cursor)
+	if err != nil {
+		return nil, err
+	}
+	// Encode cursor data string to map[string]interface{} format for futher process.
+	obj := make(map[string]interface{})
+	if err := json.Unmarshal(data, &obj); err != nil {
+		return nil, err
+	}
+
+	cursorVals := make([]interface{}, 0, len(opts.Orders))
+	for _, o := range opts.Orders {
+		val, ok := obj[o.Field]
```

Comment:
From https://github.com/pipe-cd/pipe/pull/1873#discussion_r615544624
I'm guessing `o.Field` is in a camel case but `obj`'s keys are in a snake case. Do we have to handle this diff right?

Reply:
This `obj` is unmarshaled of `o.Cursor` which be created from [iterator.go/Cursor](https://github.com/pipe-cd/pipe/pull/1873/files#diff-e3e2a6865fc4ff3cef1381ee07a923954fc7bfc5e0365e8da0941d779830cc05R58) which in fact is JSON with 🐫 case key 😁 
I wrapped all 🐍 case possible JSON in `iterator.go` so we don't need to handle that here 👀

Reply:
Ah that makes sense. sorry to interrupt. I was a little bit confused.

Reply:
No worry, your questions help a lot to review my thought 🙏 Thanks 😁